rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/helmfile/HelmfileTemplateUtils.java

```diff
@@ -0,0 +1,157 @@
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class HelmfileTemplateUtils {
+  private static final String MANIFEST_SEPARATOR = "---\n";
+  private static final Pattern REGEX_TESTS_MANIFESTS =
+      Pattern.compile("# Source: .*/templates/tests/.*");
+
+  private final ArtifactDownloader artifactDownloader;
+  private final RoscoHelmfileConfigurationProperties helmfileConfigurationProperties;
+  private final RoscoHelmConfigurationProperties helmConfigurationProperties =
+      new RoscoHelmConfigurationProperties();
+
+  public HelmfileTemplateUtils(
+      ArtifactDownloader artifactDownloader,
+      RoscoHelmfileConfigurationProperties helmfileConfigurationProperties) {
+    this.artifactDownloader = artifactDownloader;
+    this.helmfileConfigurationProperties = helmfileConfigurationProperties;
+  }
+
+  public BakeRecipe buildBakeRecipe(
+      BakeManifestEnvironment env, HelmfileBakeManifestRequest request) throws IOException {
+    BakeRecipe result = new BakeRecipe();
+    result.setName(request.getOutputName());
+
+    Path helmfileFilePath;
+
+    List<Path> valuePaths = new ArrayList<>();
+    List<Artifact> inputArtifacts = request.getInputArtifacts();
+    if (inputArtifacts == null || inputArtifacts.isEmpty()) {
+      throw new IllegalArgumentException("At least one input artifact must be provided to bake");
+    }
+
+    log.info("helmfileFilePath: '{}'", request.getHelmfileFilePath());
+    Artifact helmfileTemplateArtifact = inputArtifacts.get(0);
+    String artifactType = Optional.ofNullable(helmfileTemplateArtifact.getType()).orElse("");
+    if ("git/repo".equals(artifactType)) {
+      env.downloadArtifactTarballAndExtract(artifactDownloader, helmfileTemplateArtifact);
+
+      // If there's no helmfile path specified, assume it lives in the root of
+      // the git/repo artifact.
+      helmfileFilePath =
+          env.resolvePath(Optional.ofNullable(request.getHelmfileFilePath()).orElse(""));
+    } else {
+      try {
+        helmfileFilePath = downloadArtifactToTmpFile(env, helmfileTemplateArtifact);
+      } catch (SpinnakerHttpException e) {
+        throw new SpinnakerHttpException(fetchFailureMessage("template", e), e);
+      } catch (IOException | SpinnakerException e) {
+        throw new IllegalStateException(fetchFailureMessage("template", e), e);
+      }
+    }
+
+    log.info("path to helmfile: {}", helmfileFilePath);
+
+    try {
+      // not a stream to keep exception handling cleaner
+      for (Artifact valueArtifact : inputArtifacts.subList(1, inputArtifacts.size())) {
+        valuePaths.add(downloadArtifactToTmpFile(env, valueArtifact));
+      }
+    } catch (SpinnakerHttpException e) {
+      throw new SpinnakerHttpException(fetchFailureMessage("values file", e), e);
+    } catch (IOException | SpinnakerException e) {
+      throw new IllegalStateException(fetchFailureMessage("values file", e), e);
+    }
```

Comment:
From here up, this basically dupilcates [HelmTemplateUtils.buildBakeRecipe](https://github.com/spinnaker/rosco/blob/d49a43507428ebd67bd0942f1050c755e45ab6bc/rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/helm/HelmTemplateUtils.java#L41).  How can we share the code?

Reply:
You are right. The `HelmTemplateUtils` is spread of duplications. What about to create a common utils class for both `helm` and `helmfile` baker? Not sure under which package to place though. Perhaps I would have to create a new package `com.netflix.spinnaker.rosco.manifests.bakeutils`. what do you think?

Reply:
@dbyron-sf I've tried to make the `HelmTemplateUtils` and `HelmfileTemplateUtils` dryer reusing where I could the code. The only thing missing to reduce is https://github.com/thameezb/rosco/blob/7e1f5160aaa7459b22009484ea56c8b6dcd8aea9/rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/helmfile/HelmfileTemplateUtils.java#L55-L97 but I cannot find a good way to reduce everything when there are in the middle POJOs. do you have any better ideas?